script/prerender/manage.ts

```diff
@@ -0,0 +1,397 @@
+  const numBookPages = pages.length;
+  numPages += numBookPages;
+
+  // Wait until the work queue is ready
+  const workQueueUrl = await workQueuePromise;
+
+  console.log(`[${book.title}] Queuing ${numBookPages} book pages in batches of 10`);
+
+  for (let pageIndex = 0; pageIndex < pages.length; pageIndex += 10) {
+    const pageBatch = pages.slice(pageIndex, pageIndex + 10);
+
+    // If the entire request fails, this command will throw and be caught at the end of this file
+    // However, we also need to check if only some of the messages failed
+    const sendMessageBatchResult = await sqsClient.send(new SendMessageBatchCommand({
+      Entries: pageBatch.map((page, batchIndex) => ({
+        Id: batchIndex.toString(),
+        MessageBody: JSON.stringify({ payload: { page }, type: 'page' }),
+      })),
+      QueueUrl: workQueueUrl,
+    }));
+
+    const failedMessages = sendMessageBatchResult.Failed || [];
+    const numFailures = failedMessages.length;
+    if (numFailures > 0) {
+      throw new Error(`[SQS] [SendMessageBatch] Error: ${numFailures} out of ${pageBatch.length
+        } pages in a batch failed to be queued. Failures: ${JSON.stringify(failedMessages)}`);
+    }
+
+    const successfulMessages = sendMessageBatchResult.Successful || [];
+    const numSuccesses = successfulMessages.length;
+    if (numSuccesses < pageBatch.length) {
+      throw new Error(`[SQS] [SendMessageBatch] Unexpected response: received only ${numSuccesses
+        } successes out of ${pageBatch.length} pages in a batch but also no failures. Successes: ${
+        JSON.stringify(successfulMessages)}`);
+    }
+  }
+
+  console.log(`[${book.title}] All ${numBookPages} book pages queued`);
+
+  // console.log(`[${book.title}] Queuing sitemap`);
+
+  await sqsClient.send(new SendMessageCommand({
+    MessageBody: JSON.stringify({ payload: { pages, slug: book.slug }, type: 'sitemap' }),
+    QueueUrl: workQueueUrl,
+  }));
+
+  console.log(`[${book.title}] Sitemap queued`);
+
+  // Used in the sitemap index
+  return {
+    params: { book: { slug: book.slug } },
+    state: { bookUid: book.id, bookVersion: book.version },
+  };
+}
+
+async function manage() {
+  console.log('Preloading route components');
+
+  await Loadable.preloadAll();
+
+  archiveLoader = createArchiveLoader(REACT_APP_ARCHIVE_URL, {
+    appPrefix: '',
+    archivePrefix: ARCHIVE_URL,
+  });
+  osWebLoader = createOSWebLoader(`${OS_WEB_URL}${REACT_APP_OS_WEB_API_URL}`);
+
+  bookLoader = makeUnifiedBookLoader(archiveLoader, osWebLoader);
+
+  // We can start fetching book ToCs while the stack is created,
+  // but we need a limit so we don't use up all the memory
+  console.log(`Loading books in batches of ${MAX_CONCURRENT_BOOKS}`);
+
+  const allPagesQueuedPromise = asyncPool(
+    MAX_CONCURRENT_BOOKS, Object.entries(BOOKS), prepareAndQueueBook
+  );
+
+  // Wait for the workers stack to be ready
+  await waitUntilStackCreateComplete(
```

Comment:
I couldn't move this into createWorkersStack() because I want a failure on the create command to not trigger a stack delete command, but a failure on waitUntilStackCreateComplete() to trigger a stack delete command.

However, I extracted this and the part that gets queue urls into a separate function and rewrote a lot of this file to hopefully make everything clearer.